llvm/lib/Target/AIE/AIE2GenInstrFormats.td

```diff
@@ -18,6 +18,15 @@ class AIE2_add_r_ri_inst_alu <dag outs, dag ins,
   let alu = {mRx0,mRx,c7s,0b11,0b0};
 }
 
+class AIE2_add_r_ri_inst_alu_imm0 <dag outs, dag ins, 
+      string opcodestr = "", string argstr = "">
+    : AIE2_inst_alu_instr32 <outs, ins, opcodestr, argstr> {
```

Comment:
TODO : Explore if we can do the same without needing to define a new InstrFormat